src/rdb.c

```diff
@@ -1445,7 +1446,12 @@ ssize_t rdbSaveDb(rio *rdb, int dbid, int rdbflags, long *key_counter, unsigned
 
         initStaticStringObject(key,kvobjGetKey(kv));
         expire = kvobjGetExpire(kv);
-        if ((res = rdbSaveKeyValuePair(rdb, &key, kv, expire, dbid)) < 0) goto werr;
+        if (server.cluster_enabled && server.cluster_slot_stats_enabled)
+            oldsize = kvobjAllocSize(kv);
+        res = rdbSaveKeyValuePair(rdb, &key, kv, expire, dbid);
+        if (server.cluster_enabled && server.cluster_slot_stats_enabled)
+            updateAllocSizes(db, curr_slot, oldsize, kvobjAllocSize(kv));
```

Comment:
What are we doing here? Why do we have code as if like key size changes?

Reply:
Dicts are rehashing all the time (unless explicitly paused), during lookup and iteration.
If we don't do accounting here, it does trigger the `serverAsserts()`s that counter go to `0` in daily CI.

Reply:
if so, do all read operations in the t_hash need to update the alloc size?

Reply:
yes, I guess so.

Reply:
i'm wondering if we could make a compromise to allow a bit of a tolerance on this hash size.
Otherwise, it's a bit strange that the read operation is constantly updating the statistics.

Reply:
can't we easily catch encoding changes and use that trigger to update the stats?
in any case, i'm not worried about the stats being outdates, i'm more worried about a drift.
i.e. if we don't keep the last size with which we updated the stats, and we just use a before and after to calculate a delta, it means that if we missed updates, we're accumulating errors.